configure

```diff
@@ -5497,6 +5508,35 @@ fi
 define_in_config_if_yes "$_mpeg2" 'USE_MPEG2'
 echo "$_mpeg2"
 
+#
+# Check for libmikmod
+#
+echocheck "libmikmod"
+if test "$_mikmod" = auto ; then
+	_mikmod=no
+	cat > $TMPC << EOF
+#include <mikmod.h>
+int main(void) {
+	SAMPLE *spl = Sample_Load("abcd");
+	Sample_Free(spl);
+	return 0;
+}
+EOF
+	if test -n "$_host"; then
+		# don't execute while cross compiling
+		cc_check $MIKMOD_CFLAGS $MIKMOD_LIBS -lmikmod && _mikmod=yes
```

Comment:
Please use ` libmikmod-config` to obtain proper compilation and linker flags. See `freetype-config` in this file for a good example.